open_spiel/games/solitaire.h

```diff
@@ -0,0 +1,409 @@
+#ifndef THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+#define THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+
+#include <any>
+#include <array>
+#include <deque>
+#include <memory>
+#include <set>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include "open_spiel/spiel.h"
+#include "open_spiel/abseil-cpp/absl/types/optional.h"
+
+// An implementation of klondike solitaire:
+// https://en.wikipedia.org/wiki/Klondike_(solitaire) More specifically, it is
+// K+ solitaire, which allows the player to play any card from the deck/waste
+// that would normally become playable after some number of draws in standard
+// klondike solitaire. For a more in-depth description of K+ solitaire, see
+// http://web.engr.oregonstate.edu/~afern/papers/solitaire.pdf. This
+// implementation also gives rewards at intermediate states like most electronic
+// versions of solitaire do, rather than only at terminal states.
+
+// ANSI color codes
+inline constexpr const char *kReset = "\033[0m";
+inline constexpr const char *kRed = "\033[31m";
+inline constexpr const char *kBlack = "\033[37m";
+
+// Unicode Glyphs
+inline constexpr const char *kGlyphHidden = "\U0001F0A0";
+inline constexpr const char *kGlyphEmpty = "\U0001F0BF";
+inline constexpr const char *kGlyphSpades = "\U00002660";
+inline constexpr const char *kGlyphHearts = "\U00002665";
+inline constexpr const char *kGlyphClubs = "\U00002663";
+inline constexpr const char *kGlyphDiamonds = "\U00002666";
+inline constexpr const char *kGlyphArrow = "\U00002190";
+
+namespace open_spiel::solitaire {
+
+// Default Game Parameters =====================================================
+
+inline constexpr int kDefaultPlayers = 1;
+inline constexpr int kDefaultDepthLimit = 150;
+inline constexpr bool kDefaultIsColored = false;
+
+// Enumerations ================================================================
+
+enum SuitType {
+  kSuitNone = 0,
+  kSuitSpades,
+  kSuitHearts,
+  kSuitClubs,
+  kSuitDiamonds,
+  kSuitHidden,
+};
+enum RankType {
+  kRankNone = 0,
+  kRankA,
+  kRank2,
+  kRank3,
+  kRank4,
+  kRank5,
+  kRank6,
+  kRank7,
+  kRank8,
+  kRank9,
+  kRankT,
+  kRankJ,
+  kRankQ,
+  kRankK,
+  kRankHidden,
+};
+enum LocationType {
+  kDeck = 0,
+  kWaste = 1,
+  kFoundation = 2,
+  kTableau = 3,
+  kMissing = 4,
+};
+enum PileID {
+  kPileWaste = 0,
+  kPileSpades = 1,
+  kPileHearts = 2,
+  kPileClubs = 3,
+  kPileDiamonds = 4,
+  kPile1stTableau = 5,
+  kPile2ndTableau = 6,
+  kPile3rdTableau = 7,
+  kPile4thTableau = 8,
+  kPile5thTableau = 9,
+  kPile6thTableau = 10,
+  kPile7thTableau = 11,
+  kPileMissing = 12
+};
+
+// Constants ===================================================================
```

Comment:
Do these need to be in the header file or can we put them in the .cc in an anonymous namespace?